litgpt/eval/evaluate.py

```diff
@@ -19,6 +21,10 @@ def prepare_results(results, save_filepath, print_results=True):
         if "groups" in results:
             print(make_table(results, "groups"))
 
+    lm_version = pkg_resources.get_distribution("lm_eval").version
+    if version.parse(lm_version) >= version.parse("0.4.3"):
+        del results["config"]  # this entry is not serializable via JSON below
```

Comment:
There is another approach: 'default' argument.
```text
default: If specified, default should be a function that gets called for objects that can’t otherwise be serialized. It should return a JSON encodable version of the object or raise a TypeError. If not specified, TypeError is raised. 
```

So we can use something like
```python
json.dumps(..., default=lambda x: str(x), ...)
```

(typing from my phone)

Maybe even default = str will work

Reply:
Didn't know about this! And it seems you are right!